jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -780,143 +781,128 @@ public static String parentPath(String p)
     }
 
     /**
-     * Convert an encoded path to a canonical form.
+     * Convert a partial URI to a canonical form.
      * <p>
-     * All instances of "." and ".." are factored out.
+     * All segments of "." and ".." are factored out.
      * Null is returned if the path tries to .. above its root.
      * </p>
      *
-     * @param path the path to convert, decoded, with path separators '/' and no queries.
+     * @param path the encoded URI from the path onwards, which may contain query strings and/or fragments
      * @return the canonical path, or null if path traversal above root.
+     * @see #canonicalPath(String)
+     * @see #canonicalURI(String)
      */
-    public static String canonicalPath(String path)
+    public static String canonicalURI(String path)
```

Comment:
Is the param a path or a uri?  If it's a uri, then the param should be called `uri`. Also, there are so many decodeds, encodeds flying around, that it might be clearer to name the method, rather than just relying on the javadoc of the param.  Suggestion:  `canonicalencodedURI`.

Reply:
It is a uri. I will rename